### ISSUE TITLE
Hetzner StorageBox compatibility

### ISSUE THREAD

**Issue by Iliannnn** [Ravinou/borgwarehouse#196]
Is it possible to add a Hetzner StorageBox as a BorgWarehouse repo? We backup from a Hetzner server to the StorageBox over SSH. You can't run BorgWarehouse natively on the StorageBox because of the limited set of commands:
```
+-----------------------------------------------------------------------------+
| The following commands are available:                                       |
|   ls                                  list directory content                |
|   tree                                list directory content                |
|   cd                                  change current working directory      |
|   pwd                                 show current working directory        |
|   mkdir                               create new directory                  |
|   rmdir                               delete directory                      |
|   du                                  disk usage of files/directories       |
|   df                                  show disk usage                       |
|   dd                                  read and write files                  |
|   cat                                 output file content                   |
|   touch                               create new file                       |
|   cp                                  copy files/directories                |
|   rm                                  delete files/directories              |
|   unlink                              delete file/directory                 |
|   mv                                  move files/directories                |
|   chmod                               change file/directory permissions     |
|   md5|sha1|sha256|sha512              create hash sum of file               |
|   md5sum|sha1sum|sha256sum|sha512sum  create hash sum of file               |
|   head                                show first lines of file              |
|   tail                                show last lines of file               |
|   grep                                search for specific string in files   |
|   stat                                stat files/directory                  |
|                                                                             |
| Available as server side backend:                                           |
|   borg                                                                      |
|   rsync                                                                     |
|   scp                                                                       |
|   sftp                                                                      |
|   rclone serve restic --stdio                                               |
|                                                                             |
| Please note that this is only a restricted shell which do not               |
| support shell features like redirects or pipes.                             |
|                                                                             |
| You can find more information in our Docs:                                  |
|   https://docs.hetzner.com/robot/storage-box/                               |
+-----------------------------------------------------------------------------+
```

Is it possible to add an external repo to BorgWarehouse and therefore make this work? Thank you in advance.

**Comment by Zyztem** [comment]
+1

**Comment by MilesTEG1** [comment]
+1

**Comment by Ravinou** [comment]
Hello. I just got an Hetzner storage box (thanks to @dhenry123 for this sponsoring) and I was able to make backups to this storage very easily.

1- I installed SSHFS (as recommended in the Hetzner documentation) on the machine running my borgwarehouse container.
2- I mounted the directory on the "repo" volumes of the docker container
```
sshfs myuser@myuser.your-storagebox.de:/Hetzner-BorgWarehouse ~/volumes/repos
```
3- I restarted the container

Everything's working, I've been able to backup to the Hetzner storage box. I even moved some old repositories and that worked too.

**Comment by Iliannnn** [comment]
I will try thanks

**Comment by Ravinou** [comment]
Hmmm, I seem to have a creative error in another context. I'll keep trying.

**Comment by Ravinou** [comment]
If you have this error on launching the container : 
```
Error response from daemon: error while creating mount source path '~/volumes/repos': mkdir XXXX/XXXX/volumes/repos: file exists
```

You can solve this by using the option "allow_root" option with sshfs to authorize the docker daemon to use the mount point : 

```
sshfs -o  uid=1001,gid=1001,allow_root myuser@myuser.your-storagebox.de:/Hetzner-BorgWarehouse ~/volumes/repos
```

Everything works great and Hetzner storage box is pretty fast.

**Comment by Ravinou** [comment]
As several people have confirmed that it has worked properly since version 2.2.1, I'm closing this issue. Documentation for external storage [is available here](https://borgwarehouse.com/docs/admin-manual/external-storage/).

**Comment by faktor73** [comment]
Amazing, thanks for this helpful info @Ravinou 

I set up BorgWarehouse today for the first time and was in need to add a Hetzner StorageBox as external backup location.


For reference: I use a local NAS and had to mount the Hetzner StorageBox like this. I was logged in with my standard user on NAS, so sudo was used to mount:

sudo sshfs -o uid=1000,gid=10,allow_other,default_permissions,reconnect -p 23 USER@USER.your-storagebox.de:borgbackup-folder ~/HetznerStoragebox-Mount


Info 1:
a) On the Storebox I've created a folder named "borgbackup-folder" (name as you like).
     In the structure of a StorageBox, the path to this folder is always: "/home/borgbackup-folder"
b) on the NAS, the folder I mount the StorageBox too is inside my standard user folder and named "HetznerStoragebox-Mount"

Important Info 2:
After login to a Hetzner Storagebox you'll always land in the folder "/home" (so this is your root every time) you cannot break out, therefore it's not necessary to use the "/home"-part as part of the folderpath behind the sshfs. Just look above and you'll see, that I just use "borgbackup-folder" behind the hostname of the storagebox.de domain.

Important Info 3:
You'll need to use port 23 to connect to a StorageBox, so I've added the parameter to the sshfs too.
